### ISSUE TITLE
*URGENT*IMPORTANT*MSF*

### ISSUE THREAD

**Issue by DeveloppSoft** [cSploit/android#205]
Hello guys,
Msf' gems packages can't be installed because ruby version is too old (~ 1.9.3).
Currently, we need to compile the last version as soon as possible.
If somebody know how to port it, please explain US; then, if somebody have write access to csploit's ruby repositorie he should merge it.
Please, trust me, I think it can work.
Sincerely, DeveloppSoft.

**Comment by Pukson** [comment]
https://rubygems.org/gems/calabash-android/versions/0.5.12 something?

**Comment by DeveloppSoft** [comment]
Hi @Pukson and thanks again :smile: 
That is not what I search, the goal is to port a new ruby version, like here : https://github.com/cSploit/android.native.ruby, but with the last version, because that version is too old.

**Comment by DeveloppSoft** [comment]
Currently, I don't know how they did it.

**Comment by Pukson** [comment]
so the link you send you want that compiled?, most schools have vacation right now right? TuxMind please give us some directions

**Comment by DeveloppSoft** [comment]
Maybe we can use the first idea of @tux-mind, the author of cSploit, using an image with ruby 2.*, and msf.
Yes , we need help.

**Comment by Pukson** [comment]
We could try it, maybe someone can get  in contact with tuxmind
Op 17 jul. 2015 6:00 PM schreef TEISSONNIERE Eliott <notifications@github.com>:Maybe we can use the first idea of @tux-mind, the author of cSploit, using an image with ruby 2.*, and msf.
Yes , we need help.

**Comment by DeveloppSoft** [comment]
But, we must compile a new version (maybe https://github.com/ruby/ruby patched for android), we need help :sos:, we are working on fixing the major problem of cSploit, and we are... two, that is not normal.

**Comment by DeveloppSoft** [comment]
I think i found his blog : http://www.tuxmind.org/

**Comment by DeveloppSoft** [comment]
...Or not

**Comment by DeveloppSoft** [comment]
I read that : https://github.com/cSploit/android/wiki/How-the-MetaSploit-Framework-is-used-in-cSploit, but it doesn't help us

**Comment by DeveloppSoft** [comment]
i will try to see the code of the linux deploy app, maybe we can use it to use an image, because I haven't find how to port ruby

**Comment by Pukson** [comment]
Go for it! But isnt it an option to put the whole metasploit image including Ruby on SD card? So that we link the app to the image..
Op 17 jul. 2015 6:17 PM schreef TEISSONNIERE Eliott <notifications@github.com>:i will try to see the code of the linux deploy app, maybe we can use it to use an image, because I haven't find how to port ruby

**Comment by DeveloppSoft** [comment]
I think we can use a tiny kali arm image with just ruby and msf, we chroot in and launch msfrpc (on lo interface).

**Comment by DeveloppSoft** [comment]
Yes, if we don't fail, users will can put the file on the sdcard, maybe we can include an option with img path

**Comment by Pukson** [comment]
Sounds possible, but without tuxmind its hard
Op 17 jul. 2015 6:21 PM schreef TEISSONNIERE Eliott <notifications@github.com>:I think we can use a tiny kali arm image with just ruby and msf, we chroot in and launch msfrpc (on lo interface).

**Comment by DeveloppSoft** [comment]
yes, I have seen it : http://hamelot.co.uk/android/run-gentoo-on-android-via-chroot/, maybe it can help us to chroot into the img file

**Comment by DeveloppSoft** [comment]
Here is the inux deploy repo : https://github.com/meefik/linuxdeploy

**Comment by Pukson** [comment]
Got the app already installed, LinuxDeploy its in the playstore. So we need to make an option in the app to direct it to the img file?
Op 17 jul. 2015 6:24 PM schreef TEISSONNIERE Eliott <notifications@github.com>:Here is the inux deploy repo : https://github.com/meefik/linuxdeploy

**Comment by DeveloppSoft** [comment]
I think we can find how they use img file use the same method

**Comment by DeveloppSoft** [comment]
See you soon @Pukson, I will make break, and post something in the next 3 days, I hope I will have found something

**Comment by fat-tire** [comment]
FWIW, if you use the [fattire-rebased](https://github.com/offensive-security/nethunter-app/tree/fattire-rebased) branch of Nethunter, it will automatically install a kali chroot for you from the UI.

**Comment by tux-mind** [comment]
give me some day guys, last exam is on 23/7.
from that day on I'm all yours, with more skills and all my love for this
project :)
i swear <3

**Comment by tux-mind** [comment]
I've also talked with the Nethunter guys time ago, I'll resume all this to
bring you the best all-in-one penetration suite in your hand ;)
please trust in me and wait a little longer ;)

**Comment by Pukson** [comment]
Awesome! TuxMind is back!!
Op 17 jul. 2015 7:46 PM schreef tux-mind <notifications@github.com>:give me some day guys, last exam is on 23/7.
from that day on I'm all yours, with more skills and all my love for this
project :)
i swear <3

**Comment by Pukson** [comment]
@tux-mind you think you can have a wokring build before 30th? Then i am going on vacation love to do some mitm ;)

**Comment by BEOWULF74** [comment]
Welcome. Tux-mind we loooooove u bro <3

**Comment by DeveloppSoft** [comment]
Hello guys,
I found 2 links that can help us :
https://wiki.debian.org/ChrootOnAndroid
http://rubiojr.rbel.co/hack/2013/01/10/installing-arch-linux-in-your-android-phone-chroot/
We just need to do the same things with a kali arm img here :
https://www.kali.org/downloads/
If you want help us to do it, just say, and I will post a skeleton class.

**Comment by fat-tire** [comment]
As I said, if you build the nethunter app using [my branch](https://github.com/offensive-security/nethunter-app/tree/fattire-rebased), kali can be installed in a chroot with a single click:

![Image 1]
(http://i.imgur.com/YL0IVAo.png)
![Image 2]
(http://i.imgur.com/Kka4v9f.png)
![Image 3]
(https://i.imgur.com/vzMDyjG.png)
![Image 4]
(https://i.imgur.com/cpJ92tp.png)

But that solution aside, @tux-mind already said he's getting back into it after his tests in three days, so I suggest you all just wait until he can bring everything up to date..

**Comment by DeveloppSoft** [comment]
But nethunter is a ROM, it is not an app.
We know tux-mind already do it, but it worked, so why not.

**Comment by DeveloppSoft** [comment]
@Fatire, can you send me the apk please, I can't compile any app for the
moment

**Comment by tux-mind** [comment]
The chroot way was the first approach used long time ago ( 2012 ).
I decided to change the way we use the MSF for the following reasons:

  - having the glibc running on your phone will sightly impact your battery
life
  - startup time was about 4 minutes on my nexus 4 ( 4 core 1.5GHz )
  - it require up to 1GB

Then I decided to use a native ruby installation with native gems. Porting
ruby to the bionic libc required a lot of work and time but when everything
was working...

  - only 200 MB required
  - about 1 minute to startup the MSF
  - using the bionic libc will make everything work as it should be

Furthermore we have full control over ruby, gems and the MSF. This means
that we can select a MSF branch or release, allowing you to decide which
version.

Have to go now. Will back soon guys, just hang on.

**Comment by DeveloppSoft** [comment]
@tux-mind, so happy to see you here, can you compile a new version of ruby
(or explain us), I think it can fix the problem.

**Comment by Pukson** [comment]
Just wait 3 more days, then he is done with his exams! Good luck tux-mind. 
Op 20 jul. 2015 12:15 PM schreef TEISSONNIERE Eliott <notifications@github.com>:@tux-mind, so happy to see you here, can you compile a new version of ruby
(or explain us), I think it can fix the problem.

**Comment by DeveloppSoft** [comment]
Yes, good luck, we need you

**Comment by tux-mind** [comment]
TEISSONNIERE Eliott I need someone that can work on the project when I
can't.
I'll be happy to explain you ( maybe to everyone by making a wiki page )
how MSF work in cSploit and how to fix/mantain/develop it.

here you can find a quick brief of how it works:
https://github.com/cSploit/android/wiki/How-the-MetaSploit-Framework-is-used-in-cSploit

thanks for your support guys, few hours ago I gave the penultimate exam.
Now I'm studying for the last one on 23/7 while working for my boss and
writing to you... I'm like the github octocat right now XD

see you soon ;)

**Comment by DeveloppSoft** [comment]
I already read the wiki, but I will do it again, I will be happy to be the
guy you can work on cSploit when you are not here.

**Comment by Pukson** [comment]
I am at your service to test etc. 
Good luck!
Op 20 jul. 2015 4:27 PM schreef TEISSONNIERE Eliott <notifications@github.com>:I already read the wiki, but I will do it again, I will be happy to be the
guy you can work on cSploit when you are not here.

**Comment by DeveloppSoft** [comment]
@Pukson of course :smile: .

**Comment by DeveloppSoft** [comment]
Hi,
I have a new idea, cSploit can download an archive containing :
- a working ruby version.
- metasploit with all the gems.
So, it is faster and easier, the app download on file, uncompress it,
and... that's all.
And if there is a metasploit or ruby update, we just have to modify the
archive.
Please say what you think about that.

**Comment by Pukson** [comment]
@tux-mind went everything good with the last exams? Good luck with everything mate

**Comment by tux-mind** [comment]
I'm back :heart:
My exams went very well, except for a project. That failed project made me lost an entire academic year.
I'm still searching a way to keep follow the master courses, even if I can't.

Anyway. Let's talk about cSploit.

@DeveloppSoft I'm glad to see that you are interested in this project. I really need someone that can help me.
I've already talked about changing the MSF implementation ( #97 ), feel free to write there about your ideas :blush:
Your idea seems good for stability, but require cSploit devs to manually update the packages. Will discuss about it on #97 .

For now let me finish what I was working on: native crash reporting.
I was developing the feature to store logcats online when you fill a bug.

After that I'll focus on fixing bugs, the first one is the MSF one.

I saw that there is a lot of open issues, I assume that there are enough logcats to find the problem and solve it. if not i'll request you additional info.

thank you all for your patience and sorry for be inactive for so long, but I had to take care of the future of my real-life.

c u soon :blush:

**Comment by Pukson** [comment]
Ah that sucks tux, but he. Its gonna be allright with you im sure about that.

**Comment by DeveloppSoft** [comment]
Yes, and very happy for you @tux-mind, please note @Pukson help me a lot.

**Comment by lief2** [comment]
Well I'm glad tux-mind's exams went well(except the project)
I think a wiki page for csploit is a very good idea. Devs that are interested in this project will be able to help more easily.

**Comment by DeveloppSoft** [comment]
Sure

**Comment by Pukson** [comment]
Hi there,

Any progress made? Will there be nightly's again?

**Comment by DeveloppSoft** [comment]
I re open an issue as tux-mind says but that's all nothing new

**Comment by SiemdeNijs** [comment]
Glad to see its still being continued, best of luck with your study Tux, love your work very much and happy to see csploit is being worked on!

**Comment by tux-mind** [comment]
i will break the current MSF update system for a while, just the time to develop the new one.

**Comment by hightechstl** [comment]
Great! Do you have an estimated time of completion?

**Comment by tux-mind** [comment]
few hours I think.
working on it right now

**Comment by hightechstl** [comment]
Happy to hear it. Sorry for the unsupportive attitudes of some. If I had
more programming knowledge I would be happy to join your team and help.
Sadly my programming knowledge is entry level at best.

**Comment by Rubenoo** [comment]
Yes I hope that you can make it work :))) goodluck with it.

**Comment by lief2** [comment]
Nice ;D

**Comment by hightechstl** [comment]
@tux-mind If there's anything that you need from me let me know. I'm here
to help in anyway I can

**Comment by Rubenoo** [comment]
@tux-mind Hey, are you making any progress? I wish you all the luck with making it work btw

**Comment by hightechstl** [comment]
Just wanted to check in and see if there was any progress with this?

**Comment by tux-mind** [comment]
I've encountered an issue.
I miss some compiled native gem like nokogiri and pcaprub.
I'm not at home right now and setting up the environment to cross compile them will require a lot of time compared to just copy and paste a file.

btw I want to share some good news with you.
including the gems into the MSF package will speed up the installation process a lot with a small price: 20/30MB of extra space that can be freed after the installation.
also we don't need a gem server.

I have the compiled gems at home, I'll get back around 5/9.
If someone have a working MSF installation it will be great if he can zip/tar the `ruby` folder and share it there, so i can extract the missing native gems.

sorry for the inconvenience guys.

**Comment by Rubenoo** [comment]
Nicceeee keep on working @tux-mind you got this. I'm so happy that were getting closer to a fix :)

**Comment by fat-tire** [comment]
hey @tux-mind--  the latest work-in-progress [nethunter apk](https://github.com/binkybear/AnyKernel2/blob/b0f65033e6dc8d42ea1cd64eee3b86902bf617b1/system/priv-app/Nethunter/nethunter.apk) beta build should auto-install those files right to `/usr/share/metasploit-framework/vendor/bundle/ruby/` on your device's chroot, including ` nokogiri-1.6.6.2` and `pcaprub-0.12.0`.  That might be the easiest way to easily access anything you need. You can probably also find them in the slightly-older [premade nethunter roms](https://www.offensive-security.com/kali-linux-nethunter-download/), but those are big and device-specific.  Still, you can dig through the chroot to find what you want.

Also kali's apt pulls from [here](http://http.kali.org/kali/pool/main/r/) (ms is [here](http://mirrors.hust.edu.cn/kali/pool/main/m/metasploit-framework), etc.) if you need sources.

**Comment by domenicoblanco** [comment]
I installed Mali NetHunter in my Nexus 6, MSF works!

**Comment by tux-mind** [comment]
thanks for your efforts @fat-tire , but is not what we need.

those gems are compiled into the chroot environment, so they use the glibc, not the bionic one.
I'll setup the cross-compile environment on my laptop when i get back home, making a nice howto into the wiki, so everything will be more clear to those great collaborators, like you, who want to help.

thanks anyway for the suggestion, really appreciated :heart:

**Comment by fat-tire** [comment]
@tux-mind Ah you're right, my bad.  Well, I've definitely cross-compiled kernels/bootloaders and android before (using CROSS_COMPILE=/path/to/compiler/arm-linux-gnueabi- ARCH=arm ) but I'm not familiar with even which source you're using.. [here](https://github.com/cSploit/android.MSF)?  Makes sense to me to just wait for you to write up the steps so I know It'll be linking to bionic properly, etc.

**Comment by Rubenoo** [comment]
@tux-mind Are you making any progress and can you work on this today?

**Comment by bandwagonz** [comment]
@Rubenoo patience is a virtue. Tux already let us know that he will resume work on the 5th. Remember that you're free to help out as well :)

**Comment by tux-mind** [comment]
working on it right now.
setting up the cross-compile environment is a very complex task... may I have to check if there is another way to achieve this.
thanks for your patience guys.

**Comment by Rubenoo** [comment]
@tux-mind no problem , do what you have to do mate. I know that you can do it

**Comment by pmcdougal** [comment]
Excited to see what you come up with as a fix :)

**Comment by tux-mind** [comment]
wow, i reach a checkpoint :blush:

I'm taking so much because I'm working on it in my little free time, but I'm doing it :+1:

I successfully build the makefile for the nokogiri gem.

the following is the command i gave, just to give you an idea of the work spent searching for missing libraries and broken argument parsing functions.

```
CONFIGURE_ARGS="--with-ldflags='-lexslt -lxslt -lxml2 -licuuc -lstdc++'" ruby -r /media/data/documents/programs/csploit/android/cSploit/jni/ruby/rbconfig.rb /data/ruby/bin/gem compile nokogiri-1.6.0.gem -- --with-zlib-include=/media/data/documents/programs/csploit/android/cSploit/jni/zlib/src --with-xslt-include=/media/data/documents/programs/csploit/android/cSploit/jni/libxslt/include --with-xml2-include=/media/data/documents/programs/csploit/android/cSploit/jni/libxml2/include --with-opt-lib=/media/data/documents/programs/csploit/android/cSploit/obj/android-9/local/armeabi --with-opt-include=/media/data/documents/programs/csploit/android/cSploit/jni/icu4c/common
```

I'll keep you posted :wink:

**Comment by tux-mind** [comment]
gotcha :v:

```
Successfully built RubyGem
  Name: nokogiri
  Version: 1.6.0
  File: nokogiri-1.6.0-arm-linux.gem
```

**Comment by sorano** [comment]
Congrats! Well done tux-mind!

**Comment by swelit** [comment]
So does it work right now?

**Comment by Rubenoo** [comment]
Wow Congrats @tux-mind Your amazing. But when can we use the fully working csploit? i cant wait btw :P

**Comment by tux-mind** [comment]
ok, the bundle is complete, I've compiled both `nokogiri` and `pcaprub` gems.

Now I have to make some test, sadly my phone is in maintenance in these days, I'll try with the android emulator.

the size of the MSF package is around 47MB ( 128MB extracted ).

**Comment by Rubenoo** [comment]
Nicce but can we test it to for now? And  send you the reports after?

**Comment by hightechstl** [comment]
I would be happy to test. Moto G, Galaxy S4, Nexus 5 32GB, as well as nexus
7 first and second gen. All I need is an apk and I can test away across
multiple devices.

**Comment by evertking** [comment]
I'm willing to test also.. S3, nexus 5 and OnePlus one.

**Comment by domenicoblanco** [comment]
I can also test on a Nexus 6 and a Samsung Galaxy Tab 3 8.0

**Comment by swelit** [comment]
I would like to test on my Samsung galaxy s5

**Comment by uranium-waffles** [comment]
Congratulations @tux-mind Nice job 😁

**Comment by tux-mind** [comment]
almost done, need to fix some little error.

```
root@generic:/mnt/media_rw/sdcard/msf # ./msfconsole                           
/mnt/media_rw/sdcard/ruby/lib/ruby/gems/1.9.1/gems/bundler-1.10.6/lib/bundler/shared_helpers.rb:78: warning: Insecure world writable dir /data/bb in PATH, mode 040777
WARNING: linker: nokogiri.so: unused DT entry: type 0xf arg 0x12dc5
WARNING: Nokogiri was built against LibXML version 2.7.8, but has dynamically loaded 20708.3.58
[*] Starting the Metasploit Framework console...WARNING: linker: pg_ext.so: unused DT entry: type 0xf arg 0x1af2
[-] WARNING! The following modules could not be loaded!
[-]     /mnt/media_rw/sdcard/msf/modules/post/multi/gather/lastpass_creds.rb: LoadError dlopen failed: cannot locate symbol "sqlite3_load_extension" referenced by "sqlite3_native.so"... - /mnt/media_rw/sdcard/ruby/lib/ruby/gems/1.9.1/gems/sqlite3-1.3.9-arm-linux/lib/sqlite3/sqlite3_native.so

/# cowsay++
 ____________
< metasploit >
 ------------
       \   ,__,
        \  (oo)____
           (__)    )\
              ||--|| *


       =[ metasploit v4.11.0-dev [core:4.11.0.pre.dev api:1.0.0]]
+ -- --=[ 1378 exploits - 773 auxiliary - 221 post        ]
+ -- --=[ 342 payloads - 37 encoders - 8 nops             ]
+ -- --=[ Free Metasploit Pro trial: http://r-7.co/trymsp ]

infocmp: couldn't open terminfo file /opt/android-ndk/platforms/android-9/arch-arm/usr/share/terminfo/v/vt100.
infocmp: couldn't open terminfo file /opt/android-ndk/platforms/android-9/arch-arm/usr/share/terminfo/v/vt100.
stty: standard input
infocmp: couldn't open terminfo file /opt/android-ndk/platforms/android-9/arch-arm/usr/share/terminfo/v/vt100.
infocmp: couldn't open terminfo file /opt/android-ndk/platforms/android-9/arch-arm/usr/share/terminfo/v/vt100.
msf > 
```

**Comment by DeveloppSoft** [comment]
So good @tux-mind :smile: !
I am looking at adding wifi cracking with an external wifi card and maybe changing the ui

**Comment by maddog9696** [comment]
Please do this DeveloppSoft Sir! If someone adds the ability to use an external WiFi card like an ALPHA adapter to Csploit, this would make the application light years ahead of other penetrating applications currently available! So exited for a working build tux! Keep up the amazing work guys!

**Comment by Megaeloelo** [comment]
I .... can't .... wait!

**Comment by maddog9696** [comment]
Same!! I'v literally been looking forward to the release day of a working Csploit for weeks now! Is this weird/nerdy!?? Lol Please hurry tux! ;D I know I'm not supposed to ask this....Buuut any estimation on a working build, like hours, days, weeks, or months @tux-mind ? Thanks again for all of your work sir! Much appreciated!

**Comment by MuradTroll** [comment]
Wow @tux-mind is a hero. He is using his time to give us a good reliable programm. Thank you and just like @maddog9696 I look forward to the fixed release of csploit. 

Just dont forget your study and stuff,ok?  ^^

**Comment by tux-mind** [comment]
Looks like android dropped `sqlite3_enable_load_extension` function from the sqlite library starting from lollipop.
I have to change the way we use the sqlite library in cSploit native stuff, not a long task anyway.

After this fix the MSF package should be ready.

I cannot give you an ETA, but from 5 October I start attending my master degree courses. this means that I will have much more free time to spend for cSploit.

For the wireless feature I remember that we already talked about it in another issue.
cSploit is born with that feature as TODO, sadly I haven't got enough time for implementing it.
BTW every part of the core has been designed to support this feature and we will certainly develop it.

@DeveloppSoft basically we need a program that:

  - list wifi-capable interfaces ( iwconfig/iw/airmon-ng )
  - try to enable monitor and promiscous mode ( airmon-ng )
  - give a easy to parse list of found networks ( airodump-ng + wash )
  - deauth client for sniffing WPA handshakes ( aireplay-ng )
  - send fake packets to generate IVs ( aireplay-ng )
  - sniff raw 802.11 packets ( airodump-ng )
  - crack sniffed IVs ( aircrack-ng )
  - bruteforce WPS ( reaver )

I'll personally prefer to convert the aircrack suite into many small libraries rather than get crazy on parsing an interactive cli program output.

as pointed out by @Avamander in #52 this solution is clean but require too much work.
on the other side displaying a console on a touch device is something to avoid as much as possible IMHO.

If you are interested contact me via email, I'm sure you'll find it somewhere in the internet :wink:

**Comment by DeveloppSoft** [comment]
@tux-mind, at the moment, I am adding RouterKeygen

**Comment by tux-mind** [comment]
:heart:

**Comment by Rubenoo** [comment]
♡ <3😘❤

**Comment by doringr** [comment]
I need to ask you guys one thing what knowledge do i need to have so i can help. (Like progrming, i dont know).

**Comment by MuradTroll** [comment]
@doringr app dev skills and programming knowledge (probably java or c#). Also msf knowledge would be good. All other aspects knows @tux-mind

**Comment by Rubenoo** [comment]
You have to know ruby to, I quess so

**Comment by gainan** [comment]
@doringr , besides programming knowledges, you still can help by testing pull requests, versions and new ideas. feedback about all of this is also very important. You'll have to install android studio, learn how to import github repositories , apply patches, change between branches, and compile the app to run it on your mobile.

**Comment by doringr** [comment]
OK thanks for the replies i really apreciate it, going to start practicing now.

**Comment by tux-mind** [comment]
@doringr thanks for your efforts, really appreciated.
you can contribute by helping in tens ways, from translating the app on transifex on finding duplicate issues.

@gainan thanks for your help, it's nice to have you onboard.

**Comment by tux-mind** [comment]
I just found the problem why i cannot cross compile gems on my notebook, now I can fix the sqlite problem, just give me some hour.

**Comment by MuradTroll** [comment]
@tux-mind I dunno if i am late but is the website layout new? Have u made it? Looks nice, kind of sleek ;)

**Comment by tux-mind** [comment]
@MuradTroll say thanks to @DominikTV , a new cSploit contributor :heart:

**Comment by Tajnymag** [comment]
Thanks @DominikTV :+1: :smiley:

**Comment by DominikTV** [comment]
No problem :wink:

**Comment by MuradTroll** [comment]
Seid ihr deutsch? Are you german @DominikTV ?

**Comment by DominikTV** [comment]
@MuradTroll Jo :+1:

**Comment by Rubenoo** [comment]
@DominicTV ich spreke eine bischen deutsch :P

**Comment by xaitax** [comment]
Noch einer hier. :)

**Comment by DominikTV** [comment]
@Rubenoo Nice :wink: 
@xaitax Sind so einige hier :)

**Comment by Tajnymag** [comment]
@DominikTV @MuradTroll @Rubenoo Ich spreche auch deutsch, aber ich möchte es nicht als die haupt Sprache. Pls, guys :D

**Comment by MuradTroll** [comment]
@DominikTV cool dacht ich wär hier schonder einzige. Erwartet man aber nicht bei meinem namen, stimmts ;)

**Comment by tux-mind** [comment]
WTF ?

**Comment by DominikTV** [comment]
I think we should come back to the toppic :wink:

**Comment by Rubenoo** [comment]
so @tux-mind making any progress on the sqlite problem?

**Comment by tux-mind** [comment]
much love :heart:


```
root@mako:/data/media/0/msf # ./msfconsole                                     
WARNING: linker: nokogiri.so: unused DT entry: type 0xf arg 0x12dc5
WARNING: Nokogiri was built against LibXML version 2.7.8, but has dynamically loaded 20708.3.58
[*] Starting the Metasploit Framework console...-WARNING: linker: pg_ext.so: unused DT entry: type 0xf arg 0x1af2
/

  Metasploit Park, System Security Interface
  Version 4.0.5, Alpha E
  Ready...
  > access security
  access: PERMISSION DENIED.
  > access security grid
  access: PERMISSION DENIED.
  > access main security grid
  access: PERMISSION DENIED....and...
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!
  YOU DIDN'T SAY THE MAGIC WORD!

```

**Comment by MuradTroll** [comment]
@tux-mind 
Please?
Su?
Sudo?
Do it or I stab u?
Pls do it I worked so hard?

WHAT IS THE MAGIC WORD?

**Comment by tux-mind** [comment]
ahaha, please wait, is not ready again, have to test the install process of the app now.
the screen was about a manual installation.

**Comment by MuradTroll** [comment]
@tux-mind 
Try: 
/.msfconsole --please

**Comment by Rubenoo** [comment]
Haha you guys are hilarious, try /.msfconsole hackyourownmagicword...please.

**Comment by MuradTroll** [comment]
/.msfconsole -fuckThisShitIDoWhatIWant

**Comment by Rubenoo** [comment]
^_^

**Comment by MuradTroll** [comment]
This picture pretty much sums up the problem

http://mscrmblog.net/wp-content/uploads/2009/07/powerofsudo1.jpg

**Comment by Megaeloelo** [comment]
https://github.com/rapid7/metasploit-framework/issues/4068 Maybe this can help .

**Comment by tux-mind** [comment]
Hem.... That wasn't an issue...

I shared that terminal snapshot to show you that the sqlite problem is
solved.
The msfconsole show some pretty funny things on boot, as the cow in the
first screen and the magic word in the second.

That's all.

Thanks anyway for your efforts ;)

**Comment by Rubenoo** [comment]
@tux-mind Now that the sqlite problem is solved, are you close to make msf work in csploit? anyway good luck with making it work :+1: :smiley:

**Comment by tux-mind** [comment]
done, branch `msf_repo` IS WORKING.

now let me make a release and an apk.

**Comment by maddog9696** [comment]
OMG so like now ?...in area minutes??? Fingers crossedD :

**Comment by maddog9696** [comment]
And @tux-mind; I would love to translate your website and fix up some of the English! How do I go about helping you with this sir? You mentioned transifex?

**Comment by tux-mind** [comment]
@maddog9696 for translating the website make pull requests to [this repo](https://github.com/cSploit/cSploit.github.io).

[transifex](https://www.transifex.com/csploit) is used for the app, not the website. you can help also there if you wish.

thanks in advance for any help :blush:

**Comment by tux-mind** [comment]
fixed in [v1.5.4](https://github.com/cSploit/android/releases/tag/v1.5.4)
